lib/razorpay/subscription.rb

```diff
@@ -0,0 +1,36 @@
+require 'razorpay/request'
+require 'razorpay/entity'
+
+module Razorpay
+  # Subscription API allows you to create and
+  # manage subscriptions with Razorpay
+  class Subscription < Entity
+    def self.request
+      Razorpay::Request.new('subscriptions')
+    end
+
+    def self.create(options)
+      request.create options
+    end
+
+    def self.fetch(id)
+      request.fetch id
+    end
+
+    def self.all(options = {})
+      request.all options
+    end
+
+    def self.cancel(id, options = {})
+      request.post "#{id}/cancel", options
+    end
+
+    def self.addons(id, addons)
```

Comment:
`Razorpay::Subscription.addons id, {}` looks like it's imitating the REST API structure, but this is not desirable as the naming makes it appear as a getter function, but it is actually creating an entity.
Please change the usage to `Razorpay::Addon.create subscription_id, {}`, since we've been using `create` in such a fashion across this gem.

Reply:
Changes done as desired.

_Rationale_ to have it in subscriptions was based on the url grouping `/subscriptions/:id/addons` but you are right, contextually and conceptually, above looks better.

**Note:** Added a comment on top of `Razorpay::Addon.create` with the thought to guide future contributors just in case. Can you please see and suggest what am I doing wrong or missing ?

```
# lib/razorpay/addon.rb
require 'razorpay/request'
require 'razorpay/entity'

module Razorpay
  # Addon API allows you to fetch and delete
  # subscription-addons with Razorpay
  class Addon < Entity
    def self.request
      Razorpay::Request.new('addons')
    end

    def self.fetch(id)
      request.fetch id
    end

    # request.post "/subscriptions/#{subscription_id}/addons", options
    ## results in wrong url /v1/addons//subscriptions/sub_9UIyKhqF0C64AM/addons
    def self.create(subscription_id, options)
      r = request
      r.request :post, "/subscriptions/#{subscription_id}/addons", options
    end
  end
end
```

Reply:
@harman28 let me know if any changes needed and above comment should stay or to be removed